pkg/controllers/cloud/account_poller.go

```diff
@@ -372,3 +345,107 @@ func updateCloudDiscoveredFieldsOfVirtualMachineStatus(current, discovered *clou
 func updateAccountStatus(current, discovered *cloudv1alpha1.CloudProviderAccountStatus) {
 	current.Error = discovered.Error
 }
+
+func (p *accountPoller) createVirtualMachineCR(vm *cloudv1alpha1.VirtualMachine) (err error) {
+	err = controllerutil.SetControllerReference(p.selector, vm, p.scheme)
+	if err != nil {
+		p.log.Info("error setting controller owner reference", "err", err)
+		return err
+	}
+	// save status since create will update vm object and remove status field from it
+	vmStatus := vm.Status
+	err = p.Client.Create(context.TODO(), vm)
+	if err != nil {
+		p.log.Info("virtual machine create failed", "name", vm.Name, "err", err)
+		return err
+	}
+	vmStatus.Agented = p.isVMAgented(vm)
+	vm.Status = vmStatus
+	p.log.V(0).Info("createVirtualMachineCR updating vm.Status", "vm.Status", vm.Status)
+	err = p.Client.Status().Update(context.TODO(), vm)
+	if err != nil {
+		p.log.Info("virtual machine status update failed", "account", p.namespacedName, "name", vm.Name, "err", err)
+		return err
+	}
+	return nil
+}
+
+func (p *accountPoller) updateVirtualMachineCR(vm *cloudv1alpha1.VirtualMachine) (err error) {
+	vmNamespacedName := types.NamespacedName{
+		Namespace: vm.Namespace,
+		Name:      vm.Name,
+	}
+	currentVM := &cloudv1alpha1.VirtualMachine{}
+	err = p.Get(context.TODO(), vmNamespacedName, currentVM)
+	if err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			p.log.Info("unable to find to update", "vm-name", vm.Name)
+			return err
+		}
+	}
+	// Check if status is changed from Agented to Agentless or vice-versa.
+	if currentVM.Status.Agented != p.isVMAgented(currentVM) {
+		p.log.Info("vm agent status changed", "vm-name", currentVM.Name)
+		// Delete the old VM CR, so that any dependent CR's will be deleted.
+		err = p.deleteVirtualMachineCR(currentVM)
+		if err != nil {
+			return err
+		}
+		// Create a new VM CR.
+		// Set the resource version empty, otherwise create will fail.
+		vm.ResourceVersion = ""
+		err = p.createVirtualMachineCR(vm)
+		if err != nil {
+			return err
+		}
+	} else {
+		updateCloudDiscoveredFieldsOfVirtualMachineStatus(&currentVM.Status, &vm.Status)
+		err = p.Client.Status().Update(context.TODO(), currentVM)
+		if err != nil {
+			p.log.Info("virtual machine status update failed", "account", p.namespacedName, "name", vm.Name, "err", err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *accountPoller) deleteVirtualMachineCR(vm *cloudv1alpha1.VirtualMachine) (err error) {
+	err = p.Delete(context.TODO(), vm)
+	if err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			p.log.Info("unable to delete", "vm-name", vm.Name)
```

Comment:
Rather than completely removing it, shall we change it to DEBUG level, cause we might want to know why the delete is failing, I will update the log message to include err as well.